lib/open_api_spex/cast/string.ex

```diff
@@ -1,60 +1,79 @@
 defmodule OpenApiSpex.Cast.String do
-  @moduledoc false
-  alias OpenApiSpex.Cast
+  @moduledoc """
 
-  def cast(%{value: value} = ctx) when is_binary(value) do
-    case cast_binary(ctx) do
-      {:cast, ctx} -> cast(ctx)
-      result -> result
-    end
-  end
+  This module will cast a binary to either an Elixir DateTime or Date. Otherwise it will
+  validate a binary based on maxLength, minLength, or a Regex pattern passed through the
+  schema struct.
 
-  def cast(ctx) do
-    Cast.error(ctx, {:invalid_type, :string})
-  end
+  """
+  alias OpenApiSpex.{Cast, Cast.Error}
 
-  ## Private functions
+  @schema_fields [:maxLength, :minLength, :pattern]
 
-  defp cast_binary(%{value: value, schema: %{format: :"date-time"}} = ctx)
-       when is_binary(value) do
+  def cast(%{value: value, schema: %{format: :date}} = ctx) when is_binary(value) do
+    case Date.from_iso8601(value) do
+      {:ok, %Date{} = date} ->
+        {:ok, date}
+      _ -> Cast.error(ctx, {:invalid_format, :date})
+    end
+  end
+
+  def cast(%{value: value, schema: %{format: :"date-time"}} = ctx) when is_binary(value) do
     case DateTime.from_iso8601(value) do
-      {:ok, %DateTime{}, _offset} -> Cast.success(ctx, :format)
+      {:ok, %DateTime{} = datetime, _offset} ->
+        {:ok, datetime}
       _ -> Cast.error(ctx, {:invalid_format, :"date-time"})
     end
   end
 
-  defp cast_binary(%{value: value, schema: %{format: :date}} = ctx) do
-    case Date.from_iso8601(value) do
-      {:ok, %Date{}} -> Cast.success(ctx, :format)
-      _ -> Cast.error(ctx, {:invalid_format, :date})
-    end
+  def cast(%{value: value} = ctx) when is_binary(value) do
+    apply_validation(ctx, @schema_fields)
   end
 
-  defp cast_binary(%{value: value, schema: %{pattern: pattern}} = ctx) when not is_nil(pattern) do
-    if Regex.match?(pattern, value) do
-      Cast.success(ctx, :pattern)
+  def cast(ctx) do
+    Cast.error(ctx, {:invalid_type, :string})
+  end
+
+  ## Private functions
+
+  defp apply_validation(%{value: value, schema: %{maxLength: max_length}} = ctx, [:maxLength|fields])
```

Comment:
Would you mind applying `mix format` to this file?

Reply:
I like the rewrite of the multi-try validation!

Reply:
Unfortunately, it doesn't appear that the formatter is setup for this project. I can add it as a separate PR but I have a feeling that it's going to affect way more that this one file

Reply:
It's setup in the project: https://github.com/open-api-spex/open_api_spex/blob/master/.formatter.exs

Perhaps your branch is based off an older version of master?

Reply:
Yeah, not in mine. I'll look into it

Reply:
It's not in the fork which is where my branch is based off

Reply:
Formatter may be set up but the setup is incorrect and still doesn’t appear to have been run. Phoenix is being imported but is only a dep in test environment. The format function won't run with it in the list.